com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -642,6 +650,26 @@ private static void mainInternal(String[] args) throws IOException, CompileExcep
             project.setOption("use-vanilla-lua", "true");
         }
 
+        if (cmd.hasOption("archive-resource-padding")) {
+            String resourcePaddingStr = cmd.getOptionValue("archive-resource-padding");
+            int resourcePadding = 0;
+            try {
+                resourcePadding = Integer.parseInt(resourcePaddingStr);
+            } catch (Exception e) {
+                System.out.printf("Could not parse --archive-resource-padding='%s' into a valid integer\n", resourcePaddingStr);
+                System.exit(1);
+                return;
+            }
+
+            if (!isPowerOfTwo(resourcePadding)) {
+                System.out.printf("Argument --archive-resource-padding='%s' isn't a power of two\n", resourcePaddingStr);
+                System.exit(1);
+                return;
+            }
+
+            project.setOption("archive-resource-padding", resourcePaddingStr);
+        }
```

Comment:
We really should improve the way we setup options, and pass options along.
E.g. validating integers, and not parsing them twice etc.